parts/kubernetesagentresourcesvmas.t

```diff
@@ -15,22 +15,31 @@
       "name": "[concat(variables('{{.Name}}VMNamePrefix'), 'nic-', copyIndex(variables('{{.Name}}Offset')))]",
       "properties": {
 {{if .IsCustomVNET}}
-	    "networkSecurityGroup": {
-		    "id": "[variables('nsgID')]"
-	    },
+        "networkSecurityGroup": {
+          "id": "[variables('nsgID')]"
+        },
 {{end}}
         "ipConfigurations": [
+          {{range $seq := loop 1 .IPAddressCount}}
           {
-            "name": "ipconfig1",
+            "name": "ipconfig{{$seq}}",
             "properties": {
+              {{if eq $seq 1}}
+              "primary": true,
+              {{end}}
               "privateIPAllocationMethod": "Dynamic",
               "subnet": {
-                "id": "[variables('{{.Name}}VnetSubnetID')]"
-             }
+                "id": "[variables('{{$.Name}}VnetSubnetID')]"
+              }
             }
           }
-        ],
+          {{if lt $seq $.IPAddressCount}},{{end}}
+          {{end}}
+        ]
+{{if not IsVNETIntegrated}}
```

Comment:
The check we are making here is whether we are using native VNET routing and addresses, or the RouteTable + AddressSpoofing that we used until now. This is really independent of whether you are using Azure CNI plugin or Azure network policy.

Consider these two scenarios:
1) CNI may also be used by other components, like Calico. You can have "CNIEnabled" and yet still be using RouteTable + AddressSpoofing. This is what Calico does today. Hence IsCNIEnabled is not suitable.

2) Similarly, IsAzureNetworkPolicy is not suitable, because you may not be using Azure network policy, yet use VNET routing. This is what I expect Calico (or other 3rd parties) will use in the future.

So the name should not be tied to CNI or AzureNetworkPolicy. This is an independent switch that just happens to be used only by Azure CNI plugin for now.